In [191]:
%load_ext autoreload
%autoreload 2

from src.base_classes.omic_data_loader import OmicDataLoader
from src.data_managers.concat import CatOmicDataManager

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [192]:
mrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74_disease/mrna",
)
mirna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74_disease/mirna_genes",
)
circrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74_disease/circrna",
)
# pirna_loader = OmicDataLoader(
#     data_dir="mds_data/splits_74/pirna",
# )
te_loader = OmicDataLoader(
    data_dir="mds_data/splits_74_disease/te_counts",
)

In [193]:
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    # "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}
odm = CatOmicDataManager(omic_data_loaders, n_splits=5)

save_folder = f"logs/mds_disease/{'_'.join(omic_data_loaders.keys())}.csv"
save_folder

'logs/mds_disease/mrna_mirna.csv'

In [194]:
# from src.evals.knn import KNNEvaluator

# knn_eval = KNNEvaluator(
#     data_manager=odm,
#     n_trials=20,
#     verbose=True,
#     params={"k_lb": 1, "k_ub": 20},
# )
# _ = knn_eval.evaluate()
# knn_eval.save_results(results_file=save_folder, row_name="KNN")

In [195]:
# from src.evals.svm import SVMEvaluator

# svm_eval = SVMEvaluator(
#     data_manager=odm,
#     n_trials=50,
#     verbose=True,
#     params={
#         "C_lb": 0.01,
#         "C_ub": 10,
#         "no_rfe": True,
#         # "rfe_step_range": (0.05, 0.2),
#         # "rfe_n_features_range": (100, 200),
#     },
#     mode="linear",
# )
# _ = svm_eval.evaluate()
# svm_eval.save_results(results_file=save_folder, row_name="Linear SVM")

In [207]:
from src.evals.xgboost import XGBoostEvaluator

xgb_eval = XGBoostEvaluator(
    data_manager=odm,
    n_trials=60,
    verbose=True,
)
_ = xgb_eval.evaluate()
xgb_eval.print_best_results()
xgb_eval.save_results(results_file=save_folder, row_name="XGBoost")

[I 2024-11-18 21:08:31,526] A new study created in memory with name: no-name-4a317b8e-d44a-4610-be78-758d6c449798
[I 2024-11-18 21:08:31,655] Trial 0 finished with value: 0.27794757275052406 and parameters: {'lambda': 0.12898708103577727, 'alpha': 0.9197732276427945}. Best is trial 0 with value: 0.27794757275052406.
[I 2024-11-18 21:08:31,784] Trial 1 finished with value: 0.5506353484062296 and parameters: {'lambda': 0.00020397119411212799, 'alpha': 0.0165821227842236}. Best is trial 1 with value: 0.5506353484062296.


New best score: 0.278
Best model performance:
Accuracy: 0.825 ± 0.030
F1 Macro: 0.452 ± 0.009
F1 Weighted: 0.746 ± 0.043
New best score: 0.551
Best model performance:
Accuracy: 0.878 ± 0.028
F1 Macro: 0.727 ± 0.142
F1 Weighted: 0.862 ± 0.041


[I 2024-11-18 21:08:31,919] Trial 2 finished with value: 0.5868005745549648 and parameters: {'lambda': 1.2167821863462274e-07, 'alpha': 3.969204157992835e-06}. Best is trial 2 with value: 0.5868005745549648.
[I 2024-11-18 21:08:32,052] Trial 3 finished with value: 0.6631836987350632 and parameters: {'lambda': 0.2547475335433851, 'alpha': 0.004549121600723068}. Best is trial 3 with value: 0.6631836987350632.


New best score: 0.587
Best model performance:
Accuracy: 0.866 ± 0.059
F1 Macro: 0.782 ± 0.093
F1 Weighted: 0.867 ± 0.063
New best score: 0.663
Best model performance:
Accuracy: 0.906 ± 0.032
F1 Macro: 0.815 ± 0.062
F1 Weighted: 0.898 ± 0.038


[I 2024-11-18 21:08:32,208] Trial 4 finished with value: 0.5350135409063966 and parameters: {'lambda': 0.0009623176333622269, 'alpha': 0.002297359996734283}. Best is trial 3 with value: 0.6631836987350632.
[I 2024-11-18 21:08:32,337] Trial 5 finished with value: 0.7581635155117249 and parameters: {'lambda': 0.048022077642464925, 'alpha': 0.00022654931648156057}. Best is trial 5 with value: 0.7581635155117249.
[I 2024-11-18 21:08:32,461] Trial 6 finished with value: 0.27794757275052406 and parameters: {'lambda': 0.0065974826872823725, 'alpha': 0.19802407923415558}. Best is trial 5 with value: 0.7581635155117249.


New best score: 0.758
Best model performance:
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067


[I 2024-11-18 21:08:32,597] Trial 7 finished with value: 0.6146245235210919 and parameters: {'lambda': 0.0001483760438633141, 'alpha': 1.3719038082220015e-08}. Best is trial 5 with value: 0.7581635155117249.
[I 2024-11-18 21:08:32,728] Trial 8 finished with value: 0.6119104221152568 and parameters: {'lambda': 1.1006065726624004e-05, 'alpha': 5.6875859852295295e-05}. Best is trial 5 with value: 0.7581635155117249.
[I 2024-11-18 21:08:32,858] Trial 9 finished with value: 0.5868005745549648 and parameters: {'lambda': 2.6091619975541745e-07, 'alpha': 2.6971808265916577e-08}. Best is trial 5 with value: 0.7581635155117249.
[I 2024-11-18 21:08:32,992] Trial 10 finished with value: 0.7172503758053912 and parameters: {'lambda': 0.009585090203027518, 'alpha': 2.4879915125083738e-06}. Best is trial 5 with value: 0.7581635155117249.
[I 2024-11-18 21:08:33,127] Trial 11 finished with value: 0.7172503758053912 and parameters: {'lambda': 0.01625284414004543, 'alpha': 2.935872461448639e-06}. Best is 

Best model performance:
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067


In [197]:
# from src.evals.mlp import MLPEvaluator

# mlp_eval = MLPEvaluator(
#     data_manager=odm,
#     n_trials=10,
#     verbose=True,
#     params={
#         "lr_range": [1e-4, 1e-2],
#         "l2_lambda": 5e-4,
#         "dropout_range": [0.1, 0.6],
#         "hidden_channels": 64,
#         "proj_dim": 64,
#         "batch_size": 32,
#         "max_epochs": 50,
#     },
# )
# _ = mlp_eval.evaluate()
# mlp_eval.save_results(results_file=save_folder, row_name="MLP")

In [198]:
# mlp_eval.print_best_results()
# mlp_eval.print_best_parameters()

In [199]:
from src.evals.mogonet import MOGONETEvaluator
from src.data_managers.sample_graph import SampleGraphDataManager

mogonet_eval = MOGONETEvaluator(
    data_manager=SampleGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "graph_style": "threshold",
            "self_connections": True,
            "avg_degree": 5,
        },
    ),
    n_trials=1,
    params={
        "encoder_hidden_channels": {
            "mrna": 64,
            "mirna": 64,
            "circrna": 64,
            "te": 64,
        },
        "encoder_type": "gat",
        "dropout": 0.2,
        "integrator_type": "vcdn", # vcdn seems to work better on mds disease
        "integration_in_dim": 16,
        "vcdn_hidden_channels": 16,
        "epochs": 250,
        "log_interval": 251,
    }
)
mogonet_eval.evaluate()
mogonet_eval.print_best_results()
mogonet_eval.save_results(results_file=save_folder, row_name="MOGONET")

[I 2024-11-18 20:57:16,415] A new study created in memory with name: no-name-66b09f7f-44f0-40b5-9dea-1ae68101c2b3


Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator


[I 2024-11-18 20:57:36,079] Trial 0 finished with value: 0.9066111158287619 and parameters: {}. Best is trial 0 with value: 0.9066111158287619.


New best score: 0.907
Best model performance:
Accuracy: 0.973 ± 0.033
F1 Macro: 0.957 ± 0.053
F1 Weighted: 0.973 ± 0.033
Best model performance:
Accuracy: 0.973 ± 0.033
F1 Macro: 0.957 ± 0.053
F1 Weighted: 0.973 ± 0.033


# linear
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
# vcdn
- integration dim = 2
Accuracy: 0.825 ± 0.030
F1 Macro: 0.452 ± 0.009
F1 Weighted: 0.746 ± 0.043
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.973 ± 0.033
F1 Macro: 0.957 ± 0.053
F1 Weighted: 0.973 ± 0.033
- integration dim = 16
Accuracy: 0.973 ± 0.053
F1 Macro: 0.958 ± 0.083
F1 Weighted: 0.973 ± 0.053
# attention - faster than vcdn
- integration dim = 2
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
- integration dim = 16
Accuracy: 0.960 ± 0.053
F1 Macro: 0.934 ± 0.085
F1 Weighted: 0.959 ± 0.054

In [202]:
from src.evals.birgat import BiRGATEvaluator
from src.data_managers.bipartite_graph import BipartiteGraphDataManager

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                "te": 1.7,
            },
        },
    ),
    params={
        "epochs": 350,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": False,
    },
    n_trials=1,
)
birgat_eval.evaluate()
birgat_eval.print_best_results()
birgat_eval.save_results(results_file=save_folder, row_name="BiRGAT 2L")

isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)


[I 2024-11-18 21:00:02,388] A new study created in memory with name: no-name-9364b903-2d4f-44c0-b9b2-6464edabea98


isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)

Epoch: 050:
Train Loss: 0.3159, Train Acc: 0.8305, Train F1 Macro: 0.4537, Train F1 Weighted: 0.7536
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111
##################################################

Epoch: 100:
Train Loss: 0.1421, Train Acc: 0.9492, Train F1 Macro: 0.8969, Train F1 Weighted: 0.9454
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026
Test Acc: 0.8667, Test F1 Macro: 0.7115, Test F1 Weighted: 0.8385
##################################################

Epoch: 150:
Train Loss: 0.1138, Train Acc: 1.0000, Train F1 Macro: 1.0000, Train F1 Weighted: 1.0000
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026

[I 2024-11-18 21:01:08,899] Trial 0 finished with value: 0.8653691107941822 and parameters: {}. Best is trial 0 with value: 0.8653691107941822.



Epoch: 350:
Train Loss: 0.0066, Train Acc: 1.0000, Train F1 Macro: 1.0000, Train F1 Weighted: 1.0000
Val Acc: 0.9286, Val F1 Macro: 0.8783, Val F1 Weighted: 0.9342, Val Geometric Mean: 0.9133
Test Acc: 0.9286, Test F1 Macro: 0.8783, Test F1 Weighted: 0.9342
##################################################
New best score: 0.865
Best model performance:
Accuracy: 0.960 ± 0.033
F1 Macro: 0.939 ± 0.051
F1 Weighted: 0.960 ± 0.033
Best model performance:
Accuracy: 0.960 ± 0.033
F1 Macro: 0.939 ± 0.051
F1 Weighted: 0.960 ± 0.033


In [201]:
# birgat_eval.save_results(results_file=save_folder, row_name="BiRGAT")
save_folder

'logs/mds_disease/mrna_mirna.csv'

```python
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                # "te": 1.8,
            },
        },
    ),
    params={
        "epochs": 250,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": False,
    },
    n_trials=1,
)
```

Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053

# mrna, mirna, circrna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, circrna, 2L no interactions
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064
---
Accuracy: 0.946 ± 0.027
F1 Macro: 0.904 ± 0.048
F1 Weighted: 0.944 ± 0.028
# mrna, mirna, circrna 3L, interactions, degree ~20 in diff exp graphs, larger degree shows degraded performance
# making the avg degree to high shows large jumps on the validation set during training
Accuracy: 0.945 ± 0.053
F1 Macro: 0.910 ± 0.081
F1 Weighted: 0.946 ± 0.048
# mrna, mirna, circrna 2L, interactions, 64 cap
Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053
# mrna, mirna, circrna 3L, interactions, 64 cap
Accuracy: 0.891 ± 0.054
F1 Macro: 0.801 ± 0.088
F1 Weighted: 0.888 ± 0.056
# mrna, mirna, circrna 3L
Accuracy: 0.920 ± 0.050
F1 Macro: 0.829 ± 0.112
F1 Weighted: 0.907 ± 0.062
# mrna, mirna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, 3L
Accuracy: 0.947 ± 0.050
F1 Macro: 0.897 ± 0.089
F1 Weighted: 0.944 ± 0.051